src/scanner.c

```diff
@@ -274,13 +274,113 @@ bool tree_sitter_gdscript_external_scanner_scan(void *payload, TSLexer *lexer,
                 break;
             }
 
+            // Store the current indentation level before processing the comment
+            uint32_t comment_indent_length = indent_length;
+            
+            // If we're in a situation where we need to emit a DEDENT first,
+            // don't consume the comment - let the DEDENT be processed first
+            if (scanner->indents->len > 0) {
+                uint16_t current_indent_length = VEC_BACK(scanner->indents);
+                if (valid_symbols[DEDENT] && indent_length < current_indent_length) {
```

Comment:
Can we move the `valid_symbols[DEDENT]` check to the outer if condition?

```suggestion
            if (valid_symbols[DEDENT] && scanner->indents->len > 0) {
                uint16_t current_indent_length = VEC_BACK(scanner->indents);
                if (indent_length < current_indent_length) {
```